src/main/java/net/dv8tion/jda/core/requests/restaction/ChannelAction.java

```diff
@@ -0,0 +1,178 @@
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.Channel;
+import net.dv8tion.jda.core.entities.EntityBuilder;
+import net.dv8tion.jda.core.entities.Guild;
+import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.entities.Role;
+import net.dv8tion.jda.core.requests.Request;
+import net.dv8tion.jda.core.requests.Response;
+import net.dv8tion.jda.core.requests.RestAction;
+import net.dv8tion.jda.core.requests.Route;
+import org.apache.http.util.Args;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONString;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @since  3.0
+ * @author Florian Spieß
+ */
+public class ChannelAction extends RestAction<Channel> //todo documentation
+{
+    public static final int ROLE_TYPE = 0;
+    public static final int USER_TYPE = 1;
+
+    protected final Set<PromisePermissionOverride> overrides = new HashSet<>();
+    protected final Guild guild;
+    protected final boolean voice;
+    protected String name;
+
+    // --voice only--
+    protected Integer bitrate = null;
+    protected Integer userlimit = null;
+
+    public ChannelAction(JDA api, Route.CompiledRoute route, String name, Guild guild, boolean voice)
+    {
+        super(api, route, null);
+        this.guild = guild;
+        this.voice = voice;
+    }
+
+    public ChannelAction setName(String name)
+    {
+        Args.notNull(name, "Channel name");
+        if (name.length() < 2 || name.length() > 100)
+            throw new IllegalArgumentException("Provided channel name must be 2 to 100 characters in length");
+
+        this.name = name;
+        return this;
+    }
+
+    public ChannelAction addPermissionOverride(Role role, Collection<Permission> allow, Collection<Permission> deny)
+    {
+        Args.notNull(role, "Override Role");
+        if (!role.getGuild().equals(guild))
+            throw new IllegalArgumentException("Specified Role is not in the same Guild!");
+
+        String id = role.getId();
+        PromisePermissionOverride override = new PromisePermissionOverride(ROLE_TYPE, id, deny, allow);
+        overrides.add(override);
+        return this;
+    }
+
+    public ChannelAction addPermissionOverride(Member member, Collection<Permission> allow, Collection<Permission> deny)
+    {
+        Args.notNull(member, "Override Member");
+        if (!member.getGuild().equals(guild))
+            throw new IllegalArgumentException("Specified Member is not in the same Guild!");
+
+        String id = member.getUser().getId();
+        PromisePermissionOverride override = new PromisePermissionOverride(USER_TYPE, id, deny, allow);
+        overrides.add(override);
+        return this;
+    }
+
+    // --voice only--
+    public ChannelAction setBitrate(Integer bitrate)
+    {
+        if (bitrate != null && bitrate < 8000)
+            throw new IllegalArgumentException("Bitrate must be greater than 8000.");
```

Comment:
The max bitrate is different for VIP guilds. I need a way to know whether the current guild is VIP or not before I can add a max check. The min always works.